src/firecracker/src/main.rs

```diff
@@ -83,6 +83,7 @@ impl From<MainError> for ExitCode {
             MainError::InvalidLogLevel(_) => FcExitCode::BadConfiguration,
             MainError::RunWithApi(ApiServerError::MicroVMStoppedWithoutError(code)) => code,
             MainError::RunWithApi(ApiServerError::MicroVMStoppedWithError(code)) => code,
+            MainError::RunWithoutApiError(RunWithoutApiError::Shutdown(code)) => code,
```

Comment:
Can we add an integration test too? to the `test_api.py::test_happy` case maybe?

Reply:
To get the behavior as seen in the issue requires the VCPU's to exit on startup, to do this requires a custom kernel/initrd that does this, as seen https://github.com/firecracker-microvm/firecracker/issues/4176#issuecomment-1764290192

Reply:
can we create a github issue to extract exit_code of Firecracker and see if someone from the community can help on it?

Reply:
Yes I can add an issue for adding the ability to get the exit code with our test framework.

Reply:
I've created #4186 .

Reply:
If main() is modified as below then we can actually verify the effect of this change:
```
fn main() -> ExitCode {
    let result = main_exec();
    if let Err(err) = result {
        error!("{err}");
        eprintln!("Error: {err:?}");
        let exit_code = ExitCode::from(err);
        error!("{exit_code:?}");
        exit_code
    } else {
        ExitCode::SUCCESS
    }
}
```
with the change the print is ExitCode(unix_exit_status(0))
while without this change the print is ExitCode(unix_exit_status(1))

Reply:
ok so as per offline discussion:
- with or without the fix we still get the print "RunWithoutApiError error: MicroVMStopped without an error: Ok" in test_config_start_no_api_exit() so the test doesn't really test the fix however, it still makes sure that we get an Ok print for shutdown case.
- This is ok for now since we have a follow up PR where we won't see the "RunWithoutApiError error: MicroVMStopped without an error: Ok" print for the shutdown case and instead will get a proper shutdown ok message.
- We don't want to go ahead with this suggestion of adding error print in main() is because it will be redundant once the follow up PR is done.